sympy/core/tests/test_assumptions.py

```diff
@@ -413,31 +456,31 @@ def test_symbol_falsepositive():
 def test_symbol_falsepositive_mul():
     # To test pull request 9379
     # Explicit handling of arg.is_positive=False was added to Mul._eval_is_positive
-    x = 2*Symbol('x', positive=False)
+    x = 2*Symbol('x', positive=False, real=True)
```

Comment:
Yes this test was failing. The problem is this:
```
In [1]: x = Symbol('x', positive=False)                                                                                           

In [2]: x.is_positive                                                                                                             
Out[2]: False

In [3]: (2*x).is_positive                                                                                                         

In [4]: print((2*x).is_positive)                                                                                                  
None
```
I guess that can be fixed by adding Mul._eval_is_positive

Reply:
Looks like master has the same problem with nonpositive:
```
In [1]: x = Symbol('x', nonpositive=False)                                                                                        

In [2]: x.is_nonpositive                                                                                                          
Out[2]: False

In [3]: (2*x).is_nonpositive                                                                                                      

In [4]: print((2*x).is_nonpositive)                                                                                               
None
```

Reply:
This is happening because `mul` object doesn't seem to have it's own `_eval_is_nonpositive` or `_eval_is_nonnegative` methods and completely depends on the `assumptions` provided by `ManagedProperties` which doesn't imply anything.
```py
>>> from sympy import symbols
>>> x = symbols('x', nonpositive=False)
>>> (2*x).is_nonpositive
>>> (2*x)._assumptions
{'nonpositive': None, 'imaginary': None, 'zero': False, 'antihermitian': None, 'complex':
 None, 'integer': None, 'rational': None, 'irrational': None, 'even': None, 'odd': None,
'finite': None, 'composite': None, 'infinite': None, 'algebraic': None, 'commutative': Tr
ue, 'negative': False, 'positive': None, 'hermitian': None, 'real': None}
>>> (2*x).assumptions0
{}
>>> x.assumptions0
{'nonpositive': False, 'negative': False, 'zero': False, 'commutative': True}
>>>
```
Needs these classes to be added.

Reply:
The Mul `_eval_is` methods would need to be reorganised to fix it for nonpositive=False. I think I've fixed the issue for positive=False though so that I can restore these tests. Just running tests locally before pushing.

Reply:
I've just pushed the change to restore these tests